<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>HomePageClass</class>
 <widget class="QMainWindow" name="HomePageClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1920</width>
    <height>1080</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>1920</width>
    <height>1080</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1920</width>
    <height>1080</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>HomePage</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="styleSheet">
    <string notr="true">background-color: #dddddd;</string>
   </property>
   <widget class="QLabel" name="background">
    <property name="geometry">
     <rect>
      <x>-10</x>
      <y>-50</y>
      <width>1921</width>
      <height>1121</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #261FB3  ;   
    color: white;                /* White text */
    border-radius: 20px;          /* Rounded corners */
    padding: 5px 10px;           /* Padding */
}
QPushButton:hover {
    background-color: #161179;   /* Darker blue on hover */
}
QPushButton:pressed {
    background-color: #0C0950;   /* Even darker when clicked */
}</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="pixmap">
     <pixmap>backgeound.jpg</pixmap>
    </property>
    <property name="scaledContents">
     <bool>true</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignmentFlag::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLineEdit" name="nodename">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>360</y>
      <width>521</width>
      <height>51</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>40</height>
     </size>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>11</pointsize>
      <bold>true</bold>
      <kerning>true</kerning>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgba(255, 255, 255, 160);  /* semi-transparent white */
color:black;
    border: 1px solid rgba(0, 0, 0, 80);
    border-radius: 10px;
    /* space for the title */</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignmentFlag::AlignCenter</set>
    </property>
    <property name="placeholderText">
     <string>Enter city name</string>
    </property>
   </widget>
   <widget class="QGraphicsView" name="graph">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>50</y>
      <width>901</width>
      <height>541</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">
   background-color: rgba(255, 255, 255, 160);    /* Light gray background */
    border: 1px solid #cccccc;      /* Soft border */
    border-radius: 10px;  border: 2px solid black;           /* Rounded corners */
    padding: 10px;                  /* Inner spacing */

</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>60</y>
      <width>541</width>
      <height>301</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>5</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QFrame {
    background-color:transparent ;      /* Light gray background */
        /* Rounded corners */
    padding: 10px;     
 border-radius: 30px; 
             /* Inner spacing */
}</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="pixmap">
     <pixmap>Screenshot_2025-05-09_221300.png</pixmap>
    </property>
    <property name="scaledContents">
     <bool>true</bool>
    </property>
    <property name="alignment">
     <set>Qt::AlignmentFlag::AlignCenter</set>
    </property>
    <property name="wordWrap">
     <bool>true</bool>
    </property>
    <property name="indent">
     <number>0</number>
    </property>
   </widget>
   <widget class="QLineEdit" name="distance">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>540</y>
      <width>250</width>
      <height>50</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>30</height>
     </size>
    </property>
    <property name="font">
     <font>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgba(255, 255, 255, 160);  /* semi-transparent white */
color:black;
    border: 1px solid rgba(0, 0, 0, 80);
    border-radius: 10px;
    /* space for the title */</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignmentFlag::AlignCenter</set>
    </property>
    <property name="placeholderText">
     <string>Distance</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="firstcity">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>480</y>
      <width>250</width>
      <height>50</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>30</height>
     </size>
    </property>
    <property name="font">
     <font>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgba(255, 255, 255, 160);  /* semi-transparent white */
color:black;
    border: 1px solid rgba(0, 0, 0, 80);
    border-radius: 10px;
    /* space for the title */</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignmentFlag::AlignCenter</set>
    </property>
    <property name="placeholderText">
     <string>1st city</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="secondcity">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>480</y>
      <width>250</width>
      <height>50</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>50</height>
     </size>
    </property>
    <property name="font">
     <font>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgba(255, 255, 255, 160);  /* semi-transparent white */
color:black;
    border: 1px solid rgba(0, 0, 0, 80);
    border-radius: 10px;
    /* space for the title */</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignmentFlag::AlignCenter</set>
    </property>
    <property name="placeholderText">
     <string>2nd city</string>
    </property>
   </widget>
   <widget class="QPushButton" name="dfs">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>720</y>
      <width>250</width>
      <height>50</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>40</height>
     </size>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #261FB3  ;   
    color: white;                /* White text */
    border-radius: 20px;          /* Rounded corners */
    padding: 5px 10px;           /* Padding */
}
QPushButton:hover {
    background-color: #161179;   /* Darker blue on hover */
}
QPushButton:pressed {
    background-color: #0C0950;   /* Even darker when clicked */
}</string>
    </property>
    <property name="text">
     <string>DFS</string>
    </property>
   </widget>
   <widget class="QPushButton" name="dijkstra">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>600</y>
      <width>250</width>
      <height>50</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>40</height>
     </size>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #261FB3  ;   
    color: white;                /* White text */
    border-radius: 20px;          /* Rounded corners */
    padding: 5px 10px;           /* Padding */
}
QPushButton:hover {
    background-color: #161179;   /* Darker blue on hover */
}
QPushButton:pressed {
    background-color: #0C0950;   /* Even darker when clicked */
}</string>
    </property>
    <property name="text">
     <string>Dijkstra</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>540</y>
      <width>250</width>
      <height>51</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>40</height>
     </size>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #261FB3  ;   
    color: white;                /* White text */
    border-radius: 20px;          /* Rounded corners */
    padding: 5px 10px;           /* Padding */
}
QPushButton:hover {
    background-color: #161179;   /* Darker blue on hover */
}
QPushButton:pressed {
    background-color: #0C0950;   /* Even darker when clicked */
}</string>
    </property>
    <property name="text">
     <string>Add Edge</string>
    </property>
   </widget>
   <widget class="QPushButton" name="deleteEdge">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>600</y>
      <width>250</width>
      <height>51</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>40</height>
     </size>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #261FB3  ;   
    color: white;                /* White text */
    border-radius: 20px;          /* Rounded corners */
    padding: 5px 10px;           /* Padding */
}
QPushButton:hover {
    background-color: #161179;   /* Darker blue on hover */
}
QPushButton:pressed {
    background-color: #0C0950;   /* Even darker when clicked */
}</string>
    </property>
    <property name="text">
     <string>Delete Edge</string>
    </property>
   </widget>
   <widget class="QPushButton" name="addnode">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>420</y>
      <width>250</width>
      <height>51</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>40</height>
     </size>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #261FB3  ;   
    color: white;                /* White text */
    border-radius: 20px;          /* Rounded corners */
    padding: 5px 10px;           /* Padding */
}
QPushButton:hover {
    background-color: #161179;   /* Darker blue on hover */
}
QPushButton:pressed {
    background-color: #0C0950;   /* Even darker when clicked */
}</string>
    </property>
    <property name="text">
     <string notr="true">Add city</string>
    </property>
    <property name="iconSize">
     <size>
      <width>16</width>
      <height>16</height>
     </size>
    </property>
   </widget>
   <widget class="QPushButton" name="deleteNode">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>420</y>
      <width>250</width>
      <height>50</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>10</width>
      <height>40</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>16777214</width>
      <height>16777215</height>
     </size>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #261FB3  ;   
    color: white;                /* White text */
    border-radius: 20px;          /* Rounded corners */
    padding: 5px 10px;           /* Padding */
}
QPushButton:hover {
    background-color: #161179;   /* Darker blue on hover */
}
QPushButton:pressed {
    background-color: #0C0950;   /* Even darker when clicked */
}</string>
    </property>
    <property name="text">
     <string>Delete city</string>
    </property>
   </widget>
   <widget class="QPushButton" name="floyd_warshall">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>780</y>
      <width>520</width>
      <height>50</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>40</height>
     </size>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>12</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #261FB3  ;   
    color: white;                /* White text */
    border-radius: 20px;          /* Rounded corners */
    padding: 5px 10px;           /* Padding */
}
QPushButton:hover {
    background-color: #161179;   /* Darker blue on hover */
}
QPushButton:pressed {
    background-color: #0C0950;   /* Even darker when clicked */
}</string>
    </property>
    <property name="text">
     <string>Floyd Warshall</string>
    </property>
   </widget>
   <widget class="QScrollArea" name="scrollArea">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>600</y>
      <width>901</width>
      <height>231</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>10</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">
   background-color: rgba(255, 255, 255, 160);    /* Light gray background */
    border: 1px solid #cccccc;      /* Soft border */
    border-radius: 10px;  border: 2px solid black;           /* Rounded corners */
    padding: 10px;                  /* Inner spacing */

</string>
    </property>
    <property name="widgetResizable">
     <bool>true</bool>
    </property>
    <widget class="QWidget" name="scrollAreaWidgetContents">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>877</width>
       <height>207</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="logOut">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>150</width>
      <height>51</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>40</height>
     </size>
    </property>
    <property name="font">
     <font>
      <pointsize>18</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #261FB3  ;   
    color: white;                /* White text */
    border-radius: 20px;          /* Rounded corners */
    padding: 5px 10px;           /* Padding */
}
QPushButton:hover {
    background-color: #161179;   /* Darker blue on hover */
}
QPushButton:pressed {
    background-color: #0C0950;   /* Even darker when clicked */
}</string>
    </property>
    <property name="text">
     <string notr="true">Log Out</string>
    </property>
    <property name="iconSize">
     <size>
      <width>16</width>
      <height>16</height>
     </size>
    </property>
   </widget>
   <widget class="QLabel" name="shortestPath">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>660</y>
      <width>521</width>
      <height>51</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel {
    background-color: #dddddd;
    color: black;                   /* Same text color */
    border: 1px solid #444;        /* Same border */
    border-radius: 10px;           /* Same rounding */
    padding: 6px 10px;             /* Same padding */
    font-size: 14px;               /* Same font size */
}</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="bfs">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>720</y>
      <width>250</width>
      <height>50</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>40</height>
     </size>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
    background-color: #261FB3  ;   
    color: white;                /* White text */
    border-radius: 20px;          /* Rounded corners */
    padding: 5px 10px;           /* Padding */
}
QPushButton:hover {
    background-color: #161179;   /* Darker blue on hover */
}
QPushButton:pressed {
    background-color: #0C0950;   /* Even darker when clicked */
}</string>
    </property>
    <property name="text">
     <string>BFS</string>
    </property>
   </widget>
   <zorder>background</zorder>
   <zorder>dfs</zorder>
   <zorder>dijkstra</zorder>
   <zorder>pushButton</zorder>
   <zorder>deleteEdge</zorder>
   <zorder>nodename</zorder>
   <zorder>graph</zorder>
   <zorder>label_3</zorder>
   <zorder>distance</zorder>
   <zorder>firstcity</zorder>
   <zorder>secondcity</zorder>
   <zorder>deleteNode</zorder>
   <zorder>addnode</zorder>
   <zorder>floyd_warshall</zorder>
   <zorder>scrollArea</zorder>
   <zorder>logOut</zorder>
   <zorder>shortestPath</zorder>
   <zorder>bfs</zorder>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
